contracts/tokens/BaseToken.sol

```diff
@@ -153,5 +164,32 @@ abstract contract BaseToken is Context, ERC721Enumerable, CommunityOwnable {
         super._beforeTokenTransfer(from, to, firstTokenId, batchSize);
     }
 
+    /**
+     * @dev See {IERC1155-safeBatchTransferFrom}.
+     */
+    function safeBatchTransferFrom(address from, address to, uint256[] memory ids, bytes memory data) public virtual {
+        if (from != _msgSender() && !isApprovedForAll(from, _msgSender())) {
+            revert BaseToken_NotAuthorized();
+        }
+        _safeBatchTransferFrom(from, to, ids, data);
+    }
+
+    function _safeBatchTransferFrom(
+        address from,
+        address to,
+        uint256[] memory ids,
+        bytes memory data
+    )
+        internal
+        virtual
+    {
+        for (uint256 i = 0; i < ids.length; ++i) {
+            uint256 id = ids[i];
+            _safeTransfer(from, to, id, data);
+        }
+        address operator = _msgSender();
+        emit TransferBatch(operator, from, to, ids);
```

Comment:
I've added this event here for completeness-sake, but want to point out that each individual `safeTransfer()` call will emit its own `Transfer` event.

@gravityblast lemme know if you think this should be removed.

Reply:
Okay then I'll remove this for now.

Reply:
Removed the event.